check.go

```diff
@@ -894,31 +918,39 @@ func newCheckDirectoryFromDirectory(dir string) (*checkDirectory, error) {
 }
 
 // add adds an item to a checkDirectory
-func (c *checkDirectory) add(path string, typeflag byte, uid, gid int, size int64, mode os.FileMode) {
+func (c *checkDirectory) add(path string, typeflag byte, uid, gid int, size int64, mode os.FileMode, mtime int64) {
```

Comment:
(Absolutely non-blocking: The UID…mTime fields seem to always come from a `*tar.Header`, so it might be possible to just pass that pointer instead of all the individual fields.

This works just fine as is.)

Reply:
There's a lot more going on in a `tar.Header`, and while I'm wary of checking most of it, I'm particularly concerned with comparing timestamps with sub-second precision, which would be the natural thing to do with a header's `ModTime` field.  As it is, ISTR one of the older Linux filesystems only being precise to within two seconds, but I couldn't find a reference confirming that for anything other than FAT.

Reply:
> There's a lot more going on in a `tar.Header`

That’s a very fair point; and adding an _entirely new_ struct would be overkill.

---

> As it is, ISTR one of the older Linux filesystems only being precise to within two seconds, 

I think that was only FAT, at least the very first Minix filesystem already had 1 second.

(POSIX requires one second… but the kernel, for a long time, didn’t even provide causality: https://bugzilla.redhat.com/show_bug.cgi?id=244697 .)

Reply:
> (POSIX requires one second… but the kernel, for a long time, didn’t even provide causality: https://bugzilla.redhat.com/show_bug.cgi?id=244697 .)

I don't have the proper words for how I'm reacting to learning this.